test/ret_web/controllers/api-internal/change_login_email_controller_test.exs

```diff
@@ -0,0 +1,74 @@
+defmodule RetWeb.ApiInternal.V1.ChangeLoginEmailControllerTest do
+  use RetWeb.ConnCase
+  import Ret.TestHelpers
+  alias Ret.Account
+
+  @dashboard_access_header "x-ret-dashboard-access-key"
+  @dashboard_access_key "test-key"
+
+  setup_all do
+    merge_module_config(:ret, RetWeb.Plugs.DashboardHeaderAuthorization, %{dashboard_access_key: @dashboard_access_key})
+
+    on_exit(fn ->
+      Ret.TestHelpers.merge_module_config(:ret, RetWeb.Plugs.DashboardHeaderAuthorization, %{dashboard_access_key: nil})
+    end)
+  end
+
+  test "new email addresses must be valid", %{conn: conn} do
+    Account.find_or_create_account_for_email("alice@reticulum.io")
+    assert Account.exists_for_email?("alice@reticulum.io")
```

Comment:
To make it clear that your are verifying the preconditions and not the post-conditions of the function under test, avoid `refute` and `assert` and just raise.

E.g.
```elixir
true = Account.exists_for_email?("alice@reticulum.io")
```

That being said, is this precondition worth verifying? It’s just testing the line before it.

Reply:
Makes sense. I don't have a lot of practice writing tests, so advice like this is really helpful. Thank you!